libraries/support/azure/service.rb

```diff
@@ -52,12 +66,16 @@ def set_reader(name, value, override)
       self
     end
 
-    def get(url:, api_version:)
+    def get(url:, api_version:, error: nil, unwrap: nil)
       confirm_configured!
 
       cache.fetch(url) do
         body = rest_client.get(url, params: { 'api-version' => api_version }).body
-        body.fetch('value', body)
+
+        error&.call(body)
+
+        return unwrap.call(body, api_version) unless unwrap.nil?
+        structify(body.fetch('value', body))
```

Comment:
There's a sort of implicit `if/else` happening here that I think might read better if it were explicit.

```ruby
def get(url:, api_version:, error: nil, unwrap: nil)
  # ...

  if unwrap.respond_to?(:call)
    unwrap.call(body, api_version)
  else
    structify(body.fetch('value', body))
  end
end